### PR TITLE
Feature/21 get projects

### PR DESCRIPTION
**Pull request by Panda-Sharp**
#21

### CODE REVIEW

**Review by jmurth1234 — APPROVED** [review]
Looks fine to me

File: src/api/projects/{id}/get.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+            } else {
+                BuildResponse(res, HttpStatus.Success, result);
+            }
+        }
+    }
+
+};
+
+export function getPrjectById(projectId: string): Promise<Project> {
```
Comment (Arlodotexe) [inline src/api/projects/{id}/get.ts]: Spelling error

File: src/api/projects/{id}/get.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+            } else {
+                BuildResponse(res, HttpStatus.Success, result);
+            }
+        }
+    }
+
+};
+
+export function getPrjectById(projectId: string): Promise<IProject> {
```
Comment (Arlodotexe) [inline src/api/projects/{id}/get.ts]: Spelling error

File: src/api/projects/{id}/get.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    return new Promise(async (resolve, reject) => {
+
+        let project: IProject;
+        Project.findByPk(projectId, { include: [{ model: User }] }).then(
+            async result => {
+                if (result) {
+                    let proj = await DbToStdModal_Project(result).catch(reject);
```
Comment (Arlodotexe) [inline src/api/projects/{id}/get.ts]: Not using correct reject interface use (`ResponsePromiseReject`)

File: src/api/projects/{id}/get.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    let id = req.params['id'];
+
+    // If someone wants the projects for a specific user, they must be authorized
+    const authAccess = validateAuthenticationHeader(req, res);
+    if (!authAccess) return;
+
+    const authenticatedDiscordId = await GetDiscordIdFromToken(authAccess, res);
+
+    if (authenticatedDiscordId) {
+        const result = await getPrjectById(id).catch(err => genericServerError(err, res));
```
Comment (Arlodotexe) [inline src/api/projects/{id}/get.ts]: To format the response object correctly, use
```catch(err => genericServerError(err, res));```
in place of
```catch((err: IRequestPromiseReject) => BuildResponse(res, err.status, err.reason))```

**Review by Arlodotexe — CHANGES_REQUESTED** [review]

File: src/api/projects/{id}/get.ts (hunk 2 above)
Comment (Panda-Sharp) [reply src/api/projects/{id}/get.ts]: fixed

File: src/api/projects/{id}/get.ts (hunk 3 above)
Comment (Panda-Sharp) [reply src/api/projects/{id}/get.ts]: maybe you inverted last 2 comments, if I understood correctly should be fixed now

File: src/api/projects/{id}/get.ts (hunk 4 above)
Comment (Panda-Sharp) [reply src/api/projects/{id}/get.ts]: maybe you inverted last 2 comments, if I understood correctly should be fixed now

**Comment by Panda-Sharp** [comment]
@Arlodotexe any news? :)

**Pull request closed without merging**